OTA-3594: link fixes and config clarity

merltron opened the pull request:
-Fixing broken links flagged by https://www.drlinkcheck.com.
-Renamed "start-intro" to "index" for consistency.
-Tidying up minor things like formatting and
duplicate text.
-Simplified procedure with or without HSM provisioning
-Clarified recommended config

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/debugging-tips.adoc:
```diff
@@ -12,11 +12,11 @@ For VS code, there is a `launch.json` provided in the `.vscode` directory.  This
 == Creating a temporary OSTree environment
 
-Try the scripts available in the link:../tests/ostree-scripts[tests/ostree-scripts] directory.
+Try the scripts available in the link:../../../../../tests/ostree-scripts[tests/ostree-scripts] directory.
```
Relative links aren't appropriate here. It should be a fully qualified link to https://github.com/advancedtelematic/aktualizr/tree/{branch or tag}/tests

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/debugging-tips.adoc:
```diff
@@ -12,11 +12,11 @@ For VS code, there is a `launch.json` provided in the `.vscode` directory.  This
 == Inspect stored info with aktualizr-info
 
-The aktualizr-info tool can be used to dump information stored in the libaktualizr database. By default, it displays basic information such as storage type, device ID, primary ECU serial and hardware ID and provisioning status. Additional information can be requested with link:../src/aktualizr_info/main.cc[various command line parameters].
+The aktualizr-info tool can be used to dump information stored in the libaktualizr database. By default, it displays basic information such as storage type, device ID, primary ECU serial and hardware ID and provisioning status. Additional information can be requested with link:../../../../../src/aktualizr_info/main.cc[various command line parameters].
```
Ditto from above.

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/debugging-tips.adoc:
```diff
@@ -57,15 +57,15 @@ Fetch both of these down. In wireshark preferences set ssl.keylog_file to point
 aktualizr can be tested against a dummy repository containing fake images.
 
-First, generate a repository using link:../src/uptane_generator/main.cc[uptane-generator] tool:
+First, generate a repository using link:../../../../../src/uptane_generator/main.cc[uptane-generator] tool:
```
Ibid.

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/debugging-tips.adoc:
````diff
@@ -57,15 +57,15 @@ Fetch both of these down. In wireshark preferences set ssl.keylog_file to point
 ```
 
-Then, serve the generated directory using a web server such as the link:../tests/fake_http_server/fake_test_server.py[fake test server].
+Then, serve the generated directory using a web server such as the link:../../../../../tests/fake_http_server/fake_test_server.py[fake test server].
````
Ibid.

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/device-cred-prov-steps.adoc:
```diff
@@ -1,6 +1,6 @@
 = Set Up Device Provisioning
 
-In this section, you'll learn how provisioning in production is different from the provisioning method that you used to get started. You'll also understand the major steps involved in provisioning devices for production.
+Learn how provisioning in production is different from the provisioning method that you used to get started. You'll also understand the major steps involved in provisioning devices for production.
```
Second sentence here is still in the imperative mood; it makes the paragraph read funny to me.

merltron replied:
Ok, reverted back to original

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/enable-device-cred-provisioning.adoc:
```diff
@@ -6,13 +6,18 @@ Once you're ready to provision devices in production, you need to build disk ima
 
 After you have flashed those images to devices, you boot the image and install the device certicate for each device. You can install the certificate to the device's fileystem or use an HSM.
```
This needs to have a sentence added along the lines of "You can also have your HSM generate a private key and certificate on the device, then sign the certificate with your fleet root key--but we don't have docs on how to do that, so for these instructions we assume you're generating the device certificates and private keys on your development computer and copying them over."

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/enable-device-cred-provisioning.adoc:
```diff
@@ -101,27 +60,25 @@ scp -P 2222 autoprov.url root@localhost:/var/sota/import/gateway.url
 ====
 +
-. _(Optional)_ When the copy operation has completed, ssh into your device and check the aktualizr log output with the following `systemd` command:
+. _(Optional)_ When the copy operation has completed, SSH into your device and check the aktualizr log output with the following `systemd` command:
```
I don't think ssh should be capitalized, generally. `ssh` stands for "Secure Shell". It's usually left lowercase because the name of the tool you use is `ssh`, lowercase, and here we're just using the name of the tool as a verb. But if we want to go for technically correct instead of common usage, it should be SSh, which is ugly and weird.

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/posix-secondaries-bitbaking.adoc:
```diff
@@ -4,7 +4,7 @@ The goal of this doc is to guide a reader on bitbaking of two type of images `pr
 
 Please, refer to link:https://uptane.github.io/uptane-standard/uptane-standard.html[the Uptane standard] in order to grok the meaning of the `Primary` and `Secondary` terms in the given context from a theoretical standpoint.
-Please, check out link:posix-secondaries.doc[this doc] to understand these terms from a practical standpoint and
+Please, check out xref:posix-secondaries.doc[this doc] to understand these terms from a practical standpoint and
```
Commas after Please here are a bit weird.

merltron replied:
that weren't me! but fixing now.

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/posix-secondaries-bitbaking.adoc:
```diff
@@ -14,7 +14,7 @@ It is assumed that a reader is familiar with Yocto and bitbaking in general as w
 
 * link:https://github.com/advancedtelematic/meta-updater/blob/master/README.adoc[meta-updater README].
-* https://docs.ota.here.com/quickstarts/qemuvirtualbox.html[How to build 'core-image-minimal' image for QEMU]
+* https://docs.ota.here.com/ota-client/dev/build-quemu.html[How to build 'core-image-minimal' image for QEMU]
```
Should be an xref

merltron replied:
Had to add some conditional logic, since xref points to an almost empty .adoc file - it just contains an include (which would be useless if viewed on Github). If viewed in Github, still HTML link is used.

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/posix-secondaries-bitbaking.adoc:
```diff
@@ -86,7 +86,7 @@ For example,
 === Specifics of bitbaking for Raspberry Pi
 
-It is assumed that a reader is a familiar with bitbaking for Raspberry Pi in general, see link:https://docs.ota.here.com/quickstarts/raspberry-pi.html[Build a Raspberry Pi image].
+It is assumed that a reader is a familiar with bitbaking for Raspberry Pi in general, see link:https://docs.ota.here.com/ota-client/dev/build-raspberry.html[Build a Raspberry Pi image].
```
Ibid.

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/posix-secondaries.adoc:
```diff
@@ -19,15 +19,15 @@ The Secondary executable is built by default, thus following build instructions
 *Configure*
 
-A default configuration of aktualizr-secondary can be found in link:../config/posix-secondary.toml[<src-root>/config/posix-secondary.toml].
+A default configuration of aktualizr-secondary can be found in link:../../../../../config/posix-secondary.toml[<src-root>/config/posix-secondary.toml].
```
Should be an absolute link.

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/posix-secondaries.adoc:
```diff
@@ -38,16 +38,15 @@ Once it's started, you should see a message saying that aktualizr-secondary is l
 *Build*
-
-The Primary executable is built by default, thus following build instructions outlined in link:../README.adoc[the repo's README] should do the trick. If the build procedure is completed successfully then aktualizr executable can be found in ``<build-dir>/src/aktualizr_primary/aktualizr``.
+The Primary executable is built by default, thus following build instructions outlined in link:../../../../../README.adoc[the repo's README] should do the trick. If the build procedure is completed successfully then aktualizr executable can be found in ``<build-dir>/src/aktualizr_primary/aktualizr``.
```
Should be an absolute link.

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/posix-secondaries.adoc:
```diff
@@ -38,16 +38,15 @@ Once it's started, you should see a message saying that aktualizr-secondary is l
 *Configure*
 
-A default configuration of aktualizr acting as Primary can be found in link:../config/sota-local-with-secondaries.toml[<src-root>/config/sota-local-with-secondaries.toml].
+A default configuration of aktualizr acting as Primary can be found in link:../../../../../config/sota-local-with-secondaries.toml[<src-root>/config/sota-local-with-secondaries.toml].
```
Ibid.

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/posix-secondaries.adoc:
```diff
@@ -38,16 +38,15 @@ Once it's started, you should see a message saying that aktualizr-secondary is l
 One configuration parameter that is worth mentioning is `[uptane]: secondary_config_file` which specifies a path
 to secondary(ies) configuration file containing input parameters for aktualizr/Primary to communicate with any secondaries.
-This is a link:../config/posix-secondary-config.json[default secondary configuration for Primary].
+This is a link:../../../../../config/posix-secondary-config.json[default secondary configuration for Primary].
```
Ibid.

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/posix-secondaries.adoc:
```diff
@@ -57,7 +56,7 @@ Configuration parameters of secondaries for Primary/aktualizr that are worth men
 * `secondaries` -  a list of Secondary TCP/IP addresses
 
-Put your credential.zip file into the current working directory or update `[provision] provision_path` in link:../config/sota-local-with-secondaries.toml[the config] so it specifies a full path to your credential file.
+Put your credential.zip file into the current working directory or update `[provision] provision_path` in link:../../../../../config/sota-local-with-secondaries.toml[the config] so it specifies a full path to your credential file.
```
Ibid.

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/recommended-clientconfig.adoc:
```diff
@@ -2,35 +2,78 @@
+|
+* **.toml file:** 
++
+`force_install_completion=true`
```
Should be false.

patrickvacek replied:
This should still be false even with the explanatory text, right?

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/troubleshooting.adoc:
```diff
@@ -44,7 +44,7 @@ GnuTLS currently requires a kernel >= 3.17 to build. If you're using an older ke
 ----
 
-This is an upstream problem; until it's fixed in upstream poky, you can link:../files/gnutls-fix.patch[download this patch] to your `poky` directory and apply it to fix the issue.
+This is an upstream problem; until it's fixed in upstream poky, you can link:../../files/gnutls-fix.patch[download this patch] to your `poky` directory and apply it to fix the issue.
```
Should be an absolute link.

merltron replied:
> Should be an absolute link.

The link was relative on the Jekyll-based site too. It was hosted on the docs server for some reason. I.e. there was literally a directory called "files" with this one patch file in it (so I replicated it on the Antora-based site). Any idea where we should put this instead?

tkfu replied:
Oops, didn't realize that. I just looked, and this has been patched upstream in every poky version. We can remove this section altogether.

tkfu replied:
In general, in Antora, we should never be using position-relative links (i.e. `../whatever`). It should almost always be an xref. If an xref doesn't work because it's linking to another catalog or module, we should either use an `ifdef::env-github[]` to make it work in both locations, or just leave the unresolvable xref if the link isn't super important. (The github renderer has a pretty reasonable failure mode there: it will just remove the link, but leave the text.)

tkfu commented on docs/ota-client-guide/modules/ROOT/pages/uptane-generator.adoc:
```diff
@@ -2,9 +2,9 @@
-. link:../src/uptane_generator/main.cc[`uptane-generator`] is a low-level tool to generate and control an Uptane repository. It can be used to manipulate the repo created with `create_repo.sh`, or it can be used entirely independently.
+. link:../../../../../src/uptane_generator/run/create_repo.sh[`create_repo.sh`] is a script to generate a new Uptane metadata repository, an OSTree repository, and all associated credentials and configuration.
+. link:../../../../../src/uptane_generator/run/serve_repo.py[`serve_repo.py`] is a script for running a minimalistic Uptane server.
+. link:../../../../../src/uptane_generator/main.cc[`uptane-generator`] is a low-level tool to generate and control an Uptane repository. It can be used to manipulate the repo created with `create_repo.sh`, or it can be used entirely independently.
```
All three of these should be absolute links.

merltron commented:
Thanks @tkfu  I think I've fixed everything. Note that the topics with the relative links were all authored by aktualizr devs, so we might have to do a bit of training (I just fixed the "relativity").
I've now used  a "aktualizr github URL" variable for the root location for non-ASCIIDOC files.

merltron commented:
PS: I refactored all content sharing between the "getstarted" and "ota-client" components. I realised that "ota-client" should be the source of all the "build-raspberry, build-qemu, ect..." procedures and "getstarted" should pull content from the main dev guide (not the other way around). That's why there are more changes in those topics. It made fixing the xrefs easier.

patrickvacek commented:
> Note that the topics with the relative links were all authored by aktualizr devs, so we might have to do a bit of training

Yes, sorry. A training or a doc about how to write the docs would be useful.

patrickvacek commented on docs/ota-client-guide/modules/ROOT/examples/sota-local.toml:
```diff
@@ -0,0 +1,15 @@
+path = "storage"
+
+[pacman]
+type = "none"
```
So do we really have to duplicate this file from `https://github.com/advancedtelematic/aktualizr/blob/master/config/sota-local.toml`? It would be really nice if we didn't have to maintain both. The one I've linked to is not actually used anywhere as far as I can tell, but as a developer, it is a convenient reference to have in the same directory as everything else. A softlink would be an option, I think.

But more importantly, this version of the config is out of date. `secondary_configs_dir` has been replaced.

merltron replied:
> So do we really have to duplicate this file 
For the time being, until https://saeljira.it.here.com/browse/OTA-3569 is done.

> secondary_configs_dir has been replaced.
OK, I'll fix in a followup PR

patrickvacek commented on docs/ota-client-guide/modules/ROOT/pages/build-quemu.adoc:
```diff
@@ -1,3 +1,38 @@
-// MC: Included from repo in Gitlab: https://main.gitlab.in.here.com/olp/edge/ota/documentation/ota-connect-docs/tree/master/docs/getstarted/modules/ROOT/pages/qemuvirtualbox.adoc
+= Build a QEMU/VirtualBox 
```
We are missing the word "image" here?

merltron replied:
Yep, I'll fix in a followup PR (this one's merged already)
